File: week1-cpp/Mathematical_Operations_on_Images/Datatype_Conversion.cpp
// Include libraries
#include <iostream>
#include "dataPath.hpp"
#include <opencv2/opencv.hpp>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main(void){
	// Read image
	Mat image = imread(DATA_PATH+"images/boy.jpg");
	
	double scalingFactor = 1/255.0;
	double shift = 0;

	//Converting from unsigned char to float(32bit)
	image.convertTo(image, CV_32FC3, scalingFactor, shift);
	
	//Converting from float to unsigned char
	image.convertTo(image, CV_8UC3, 1.0/scalingFactor, shift);
	
	return 0;
}

File: week1-python/Mathematical_Operations_on_Images/Datatype_Conversion.py
# Import libraries
import cv2
import numpy as np
from dataPath import DATA_PATH

# Read image
image = cv2.imread(DATA_PATH+"images/boy.jpg")

scalingFactor = 1/255.0

# Convert unsigned int to float
image = np.float32(image)
# Scale the values so that they lie between [0,1]
image = image * scalingFactor

#Convert back to unsigned int
image = image * (1.0/scalingFactor)
image = np.uint8(image)